Struct viewer debugging tool

kotcrab opened the pull request:
This PR adds struct viewer debugging tool. It visualizes objects data in game memory using data types and symbols fetched from your Ghidra project. It also allows to set memory breakpoints and edit field values which is helpful when reverse engineering unknown types.

This is how it looks in action:

![image](https://github.com/user-attachments/assets/3f8e962d-e1b5-4d07-82c3-6ff96cf3ace4)

The types are fetched over REST API that Ghidra exposes with my [ghidra-rest-api](https://github.com/kotcrab/ghidra-rest-api) extension. This allows for a very efficient workflow where you can edit some types in Ghidra and instantly refresh them in Struct viewer to see updated object memory layout.

## How to try it

### If you don't have a Ghidra project

I prepared mock API you can use for testing this without setting up Ghidra at all. You will just need [Mockoon](https://mockoon.com/) and the API json I prepared [here](https://github.com/user-attachments/files/17723428/ghidra-rest-api.json). With that in Mockoon do `File` -> `Open local environment`, select json file then press the green button to start the mock server.
The types and symbols are for Fate/Extra CCC, with that game booted you can open the `Struct viewer` and press the `Connect` button. Now you can start exploring the globals, for most interesting see `g_logoTask` and `g_TitleTask` (the first one will be populated when logos are visible and the second one after getting to the title screen).

### If you have a Ghidra project

Then you just need to install my [ghidra-rest-api](https://github.com/kotcrab/ghidra-rest-api) extension, start API server in Ghidra. Open the `Struct viewer` and press the `Connect` button. Note that your Ghidra image should be at `0x8804000` for globals to align.

kotcrab committed:
Add GhidraClient

kotcrab committed:
Add struct viewer debugging tool

kotcrab committed:
Support copy address and value in Struct viewer

Reorganize add breakpoint menu
Style fixes

kotcrab committed:
Struct viewer, fix build

Fix open check

hrydgard reviewed (changes requested):
Skimmed through it, code is nice, clear and self-contained, looks good.

One request: GhidraClient.h really needs a comment clarifying that this is not an official Ghidra protocol, but one exposed by your plugin. The comment should contain a link to your repo, and at a minimum a brief usage description, like the one you wrote in this PR. And I guess a link to this PR couldn't hurt.

kotcrab committed:
Add GhidraClient and ImStructViewer docs

Few more code comments and misc clean up

kotcrab commented:
Good point, added now and few more comments. I also added more setup info to the initial UI:
![image](https://github.com/user-attachments/assets/59beddaa-9688-430e-98a5-8ef62f1e9977)

Let me know if there's anything more you would like changed.

hrydgard commented:
Very nice, good idea to show it in the UI too!

Minor, we generally only use // for comments, even large blocks, but I'm not gonna block this on that, just letting you know for future changes.
